Do an editorial pass (#158)
TheBevyFlock/bevy_quickstart: docs/tooling.md

> # Recommended 3rd-Party Tools

Reviewer: ```suggestion
# Recommended 3rd-party tools
```
Was missed

> # Bevy Quickstart Recommended 3rd-Party Tools

Reviewer: I'm a fan of normal casing for headers / titles. It's what [wikipedia does](https://en.wikipedia.org/wiki/List_of_game_engines), for example.

Not a big deal either way.

Author: Can change

Author: [`84f29b2` (#158)](https://github.com/TheBevyFlock/bevy_quickstart/pull/158/commits/84f29b259581be62fa8ebdf8c76c67e2e18a1241)

## Libraries

A few libraries that the authors of this template have vetted and think you might find useful.

| Name                                                                                   | Category       | Description                           |
| -------------------------------------------------------------------------------------- | -------------- | ------------------------------------- |
| [`leafwing-input-manager`](https://github.com/Leafwing-Studios/leafwing-input-manager) | Input          | Input -> Action mapping               |
| [`bevy_mod_picking`](https://github.com/aevyrie/bevy_mod_picking)                      | Input          | Advanced mouse interaction            |
| [`bevy-inspector-egui`](https://github.com/jakobhellermann/bevy-inspector-egui)        | Debugging      | Live entity inspector                 |
| [`bevy_mod_debugdump`](https://github.com/jakobhellermann/bevy_mod_debugdump)          | Debugging      | Schedule inspector                    |
| [`avian`](https://github.com/Jondolf/avian)                                            | Physics        | Physics engine                        |

> | [`bevy_rapier`](https://github.com/dimforge/bevy_rapier)                               | Physics        | Physics engine (not ECS driven)       |

Reviewer: ```suggestion
| [`bevy_rapier`](https://github.com/dimforge/bevy_rapier)                               | Physics        | Physics engine (not ECS-driven)       |
```
Nit

Author: https://github.com/TheBevyFlock/bevy_quickstart/pull/159

- `leafwing-input-manager` and `bevy_mod_picking` are very likely to be upstreamed into Bevy in the near future.
- `bevy-inspector-egui` and `bevy_mod_debugdump` help fill the gap until Bevy has its own editor.

> - `avian` or `bevy_rapier` helps fill the gap until Bevy has its own physics engine. `avian` is easier to use, `bevy_rapier` has better performance.

Reviewer: ```suggestion
- `avian` or `bevy_rapier` helps fill the gap until Bevy has its own physics engine. `avian` is easier to use, while `bevy_rapier` is more performant.
```

- `sickle_ui` is well-aligned with `bevy_ui` and helps fill the gap until Bevy has a full collection of UI widgets.

> None of these are necessary, but they can save you a lot of time and effort.

Reviewer: Redundant?

Author: Some people are a bit allergic to third party deps out of principle in my experience, so I'd like to stress it.

Reviewer: Understandable.

## VS Code extensions

If you're using [VS Code](https://code.visualstudio.com/), the following extensions are highly recommended:

| Name                                                                                                      | Description                       |
|-----------------------------------------------------------------------------------------------------------|-----------------------------------|
| [rust-analyzer](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer)              | Rust support                      |
| [Even Better TOML](https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml)          | TOML support                      |
| [vscode-ron](https://marketplace.visualstudio.com/items?itemName=a5huynh.vscode-ron)                      | RON support                       |

> | [Dependi](https://marketplace.visualstudio.com/items?itemName=fill-labs.dependi)                          | `crates.io` dependency resolution |

Author: `crates` has been merged into the broader `Dependi`

> | [Dependi](https://marketplace.visualstudio.com/items?itemName=fill-labs.dependi)                          | `crates.io` dependency resolution |

Reviewer: I couldn't find this extension within VS Code (maybe I have to restart?), but I do see the migration notice on the `crates` repo.

Author: ![image](https://github.com/user-attachments/assets/91a48db0-d29f-4526-827f-8b69d34fda63)

Hmm, dunno

Reviewer: Won't block on this, I'll try and figure this out.
